grails-app/controllers/org/pih/warehouse/api/CycleCountApiController.groovy

```diff
@@ -45,21 +48,98 @@ class CycleCountApiController {
         BatchCommandUtils.validateBatch(command, "requests")
         List<CycleCountDto> cycleCounts = cycleCountService.startCycleCount(command)
 
-        render([data: cycleCounts] as JSON)
+        if (!params.format) {
+            params.format = "json"
+        }
+
+        withFormat {
```

Comment:
I thought you needed to set the "accept" header for this to work? (ex: `'Accept': 'application/pdf'`). But clearly I was wrong and you just need to set the "format" query param.

How is this all interacting with setting `responseType: 'blob'` on the frontend? And then with the fileDownload method with the MIME setting? (I left a similar comment on the frontend side)

Reply:
we discussed this in Slack, but I'm happy with the solution you have here using the "format" param.